### ISSUE TITLE
Incorrect error for non-total TypedDict in stub

### ISSUE THREAD

**Issue by JelleZijlstra** [google/pytype#1122]
Over on python/typeshed#6981, I got this error:
```
ParseError: 'total' allowed as classdef kwarg only for TypedDict subclasses
```
For a typeddict that inherits from another TypedDict with total=False. The bug seems to be here: https://github.com/google/pytype/blob/fe3080843be4ac6e0d69b112417b7ad068cdcfa5/pytype/pyi/classdef.py#L63. That code looks only at the direct parent classes, but it should also allow total= if one of the parents is itself a child of TypedDict.

**Comment by rchen152** [comment]
Hmm, looks like we probably need to move the metaclass check out of the parser and into a validation visitor that runs after external types have been resolved.

**Comment by rchen152** [comment]
I've just released pytype-2022.02.17, which contains this fix.

**Comment by JelleZijlstra** [comment]
Thank you!
